packages/shared/components/Search/SearchPanel.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+import { Text, Flex } from 'design';
+import { StyledPanel } from 'design/DataTable';
+import { PageIndicatorText } from 'design/DataTable/Pager/Pager';
+import InputSearch from 'design/DataTable/InputSearch';
+import { AgentFilter } from 'teleport/services/agents';
+import Toggle from 'teleport/components/Toggle';
+import Tooltip from 'teleport/components/ServersideSearchPanel/Tooltip';
+import { PredicateDoc } from 'teleport/components/ServersideSearchPanel/ServersideSearchPanel';
+
+export function SearchPanel({
```

Comment:
Excuse me but I don't understand where the issue with `<SearchPanel/>` and `<SearchPagination/>` is that you discussed above.

`packages/shared/components/Search/SearchPanel.tsx` looks almost identical to `packages/teleport/src/components/ServersideSearchPanel/ServersideSearchPanel.tsx`. The only difference is in the callbacks passed to the particular children + the one from `packages/shared` supports `showSearchBar` as a prop. But couldn't both components reuse the same visual component and pass different props?

The situation looks similar when it comes to `packages/shared/components/Search/SearchPagination.tsx` and `packages/design/src/DataTable/Pager/Pager.tsx`. `Pager` seems to support a bunch of additional props but couldn't we reuse the visual component instead of copying it?

Reply:
The issue we spoke of was using these as components to "build" the table rather than using `Table`'s prop of `serversideProps`. I don't care where we pull the components from tbh, these two components `SearchPanel` and `SearchPagination` I just created to abstract from the files they were originally in (`NewRequest` and a non-shared version of `SearchPanel`). If they exist elsewhere, I'm happy to use those too.

Reply:
I think the least we could do here is avoid introducing additional duplication by reusing those visual components. I didn't use those table components that much so without investing more time into it, it's hard for me to say what's the best course regarding the duplication of logic between different uses of data tables.

Reply:
technically, this PR didn't introduce this duplication. This component was being used in the teleport package and I just moved it to the shared package. I can dig into why they used two different panels but you can see the [changes](https://github.com/gravitational/webapps.e/compare/74d3be697bfcc31b4c69ffe024083902c516be55...31757a460bc3f73c79c0ba301d86120d4120e051) on webappse

I wish it was a bit more clear in this PR that those files were just moved, not added.